yt_dlp/extractor/rai.py

```diff
@@ -35,189 +28,187 @@ class RaiBaseIE(InfoExtractor):
-            if url_elem is None:
-                continue
+        is_live = xpath_text(relinker, './is_live', default='N') == 'Y'
+        duration = parse_duration(xpath_text(relinker, './duration', default=None))
+        media_url = xpath_text(relinker, './url[@type="content"]', default=None)
 
-            media_url = url_elem.text
+        if not media_url:
+            self.raise_no_formats('The relinker returned no media url')
 
-            # This does not imply geo restriction (e.g.
-            # http://www.raisport.rai.it/dl/raiSport/media/rassegna-stampa-04a9f4bd-b563-40cf-82a6-aad3529cb4a9.html)
-            if '/video_no_available.mp4' in media_url:
-                continue
+        # geo flag is a bit unreliable and not properly set all the time
+        geoprotection = xpath_text(relinker, './geoprotection', default='N') == 'Y'
 
-            ext = determine_ext(media_url)
-            if (ext == 'm3u8' and platform != 'mon') or (ext == 'f4m' and platform != 'flash'):
-                continue
+        ext = determine_ext(media_url)
+        formats = []
 
-            if ext == 'mp3':
-                formats.append({
-                    'url': media_url,
-                    'vcodec': 'none',
-                    'acodec': 'mp3',
-                    'format_id': 'http-mp3',
-                })
-                break
-            elif ext == 'm3u8' or 'format=m3u8' in media_url or platform == 'mon':
-                formats.extend(self._extract_m3u8_formats(
-                    media_url, video_id, 'mp4', 'm3u8_native',
-                    m3u8_id='hls', fatal=False))
-            elif ext == 'f4m' or platform == 'flash':
-                manifest_url = update_url_query(
-                    media_url.replace('manifest#live_hds.f4m', 'manifest.f4m'),
-                    {'hdcore': '3.7.0', 'plugin': 'aasp-3.7.0.39.44'})
-                formats.extend(self._extract_f4m_formats(
-                    manifest_url, video_id, f4m_id='hds', fatal=False))
-            else:
-                bitrate = int_or_none(xpath_text(relinker, 'bitrate'))
-                formats.append({
-                    'url': media_url,
-                    'tbr': bitrate if bitrate > 0 else None,
-                    'format_id': f'http-{bitrate if bitrate > 0 else "http"}',
-                })
+        if ext == 'mp3':
+            formats.append({
+                'url': media_url,
+                'vcodec': 'none',
+                'acodec': 'mp3',
+                'format_id': 'https-mp3',
+            })
+        elif ext == 'm3u8' or 'format=m3u8' in media_url:
+            fmts = self._extract_m3u8_formats(
+                media_url, video_id, 'mp4', m3u8_id='hls', fatal=False)
+            for f in fmts:
+                if not f.get('acodec'):
+                    f['acodec'] = 'mp4a'
+                if not f.get('vcodec'):
+                    f['vcodec'] = 'avc1'
```

Comment:
Isn't this sufficient instead of #5924 ?

Reply:
not really.

it does not fix #887 nor other examples provided in the #5924 discussion. It's a bit too generic to properly work and set codecs properly.

If you are not convinced by #5924 code you can scrap the force_codecs part in this PR and go along in merging the changes for now.

We can work on the changes provided in #5924 later.

Reply:
Ok, it seems I'm misunderstanding the point of `force_codecs`. let me take another look at it.

Reply:
you probably understood correctly what force_codecs does, but #5924 does also extra checks on the m3u8 manifests to properly set the codecs, and so, a generic solution like this one is not always ok.